src/solver/opamCudf.ml

```diff
@@ -732,24 +732,12 @@ module ChainSet = struct
     if Set.is_empty hds then []
     else hds :: transpose tls
 
-  (** cs1 precludes cs2 if it contains a list that is prefix to all elements of
-      cs2 *)
-  let precludes cs1 cs2 =
-    let rec list_is_prefix pfx l = match pfx, l with
-      | [], _ -> true
-      | a::r1, b::r2 when Package.equal a b -> list_is_prefix r1 r2
-      | _ -> false
-    in
-    exists (fun pfx -> for_all (fun l -> list_is_prefix pfx l) cs2) cs1
-
   let length cs = fold (fun l acc -> min (List.length l) acc) cs max_int
 end
 
 type explanation =
   [ `Conflict of string option * string list * bool
-  | `Missing of string option * string *
-                (OpamPackage.Name.t * OpamFormula.version_formula)
-                  OpamFormula.formula
+  | `Missing of string option * string * OpamFormula.t
```

Comment:
Is it worth updating this type to include in it the `'NoLongerAvailable` variant, and even keep the high level information of formula, invariant, request in it; keeping all the work of *stringification* to the `string_of_explanation` function?

Reply:
I'm not sure to understand what you mean. `` `NoLongerAvailable `` is not defined and only locally used. The change highlighted here is only cosmetic (`OpamFormula.t = (OpamPackage.Name.t * OpamFormula.version_formula) OpamFormula.formula`)